test/realm_test.dart

```diff
@@ -1394,6 +1395,90 @@ Future<void> main([List<String>? args]) async {
     expect(await realmIsCancelled, isTrue);
     expect(progressReturned, isFalse);
   });
+
+  test('Realm - local realm can be compacted', () async {
+    var config = Configuration.local([Car.schema], path: p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm"));
+    final compacted = await Realm.compact(config);
+    expect(compacted, true);
+    
+    //test the realm can be opened. This also allows the compacted realm to be deleted after the test
+    final realm = getRealm(config);
+  });
+
+  test('Realm - local realm can be compacted in worker isolate', () async {
+    var config = Configuration.local([Car.schema], path: p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm"));
+
+    final receivePort = ReceivePort();
+    await Isolate.spawn((List<Object> args) async {
+      SendPort sendPort = args[0] as SendPort;
+      final path = args[1] as String;
+      var config = Configuration.local([Car.schema], path: path);
+      final compacted = await Realm.compact(config);
+      Isolate.exit(sendPort, compacted);
+    }, [receivePort.sendPort, config.path]);
+
+    final compacted = await receivePort.first as bool;
+    expect(compacted, true);
+
+    //test the realm can be opened. This also allows the compacted realm to be deleted after the test
+    final realm = getRealm(config);
+  });
+
+  test('Realm - local encrypted realm can be compacted', () async {
+    final config = Configuration.local([Friend.schema],
+        encryptionKey: generateEncryptionKey(), path: p.join(Configuration.defaultStoragePath, "${generateRandomString(8)}.realm"));
+    final compacted = await Realm.compact(config);
+    expect(compacted, true);
+
+    //test the realm can be opened. This also allows the compacted realm to be deleted after the test
+    final realm = getRealm(config);
+  });
+
+  test('Realm - in-memory realm can be compacted', () async {
```

Comment:
We have add a test for Read-Only realm and to expect an exception.

Reply:
done